### PR TITLE
refactor: use f-strings

### PR DESCRIPTION
**Pull request by davidculley**
Migrate `printf`-style string-formatting to f-strings.

### CODE REVIEW

File: code/japanese/reading.py [hunk 1]
```diff
@@ -152,29 +152,30 @@ def reading(self, expr: str) -> str:
                 if kanji[i] != reading[i]:
                     break
                 placeL = i + 1
-            if placeL == 0:
-                if placeR == 0:
```
Comment (dae) [inline code/japanese/reading.py:156]: As placeL/R are not optional, I think comparing to 0 is a little more explicit. Not a big deal though.

File: code/japanese/reading.py [hunk 2]
```diff
@@ -152,29 +152,30 @@ def reading(self, expr: str) -> str:
...
+                        f"{reading[-placeR:]}"
                     )
             else:
-                if placeR == 0:
+                if not placeR:
                     out.append(
-                        "%s %s[%s]"
-                        % (reading[:placeL], kanji[placeL:], reading[placeL:])
+                        f"{reading[:placeL]}"
+                        f" "
```
Comment (dae) [inline code/japanese/reading.py:168]: I think some linters will complain if a format string contains no replacements.

Comment (davidculley) [reply code/japanese/reading.py]: Right, this string doesn't need the `f`. Removed it.

File: code/japanese/reading.py (hunk 1 above)
Comment (davidculley) [reply code/japanese/reading.py:156]: `not x` is more pythonic than the more explicit `x == 0`. I'll revert it if you wish.

**Pull request merged**